File: java/1207. Unique Number of Occurrences/Solution.java
class Solution {
    public boolean uniqueOccurrences(int[] arr) {
        Map<Integer, Integer> freq = new HashMap<>();
        Arrays.stream(arr)
            .forEach(number -> freq.put(number, freq.getOrDefault(number, 0) + 1));

        int uniqueCount = (int) freq.values()
            .stream()
            .distinct()
            .count();

        return freq.size() == uniqueCount;
    }
}

File: python/1207. Unique Number of Occurrences/solution.py
class Solution:
    def uniqueOccurrences(self, arr: List[int]) -> bool:
        occurences = {}

        for number in arr:
            occurences[number] = occurences.get(number, 0) + 1

        uniqueOccurences = set(occurences.values())

        return len(occurences) == len(uniqueOccurences)